src/robot/variables/finders.py

```diff
@@ -73,9 +73,9 @@ def _get_match(self, variable):
 
     def _validate_value(self, value, identifier, name):
         if identifier == '@':
-            if not is_list_like(value):
-                raise VariableError("Value of variable '%s' is not list or "
-                                    "list-like." % name)
+            if not is_iterable(value):
+                raise VariableError("Value of variable '%s' is not iterable."
+                                    % name)
```

Comment:
I don't think this particular change is a good idea. In my opinion we should still require variables that are used like `@{var}` to be list-like, not any iterable (i.e. not a string or a file). The original issue is about using something like `${var}[1]` and that working if if `${var}` is any iterable ought to be fine.

This change is also the reason for so many tests failing.

Reply:
reverted...